### PR TITLE
Feature 2244 avatar click

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/menu/Menu.jsx [hunk 1]
```diff
@@ -20,7 +20,6 @@ import fileDownload from 'js-file-download';
 import { useLocation, Link } from 'react-router-dom';
 import { AppContext } from '../../context/AppContext';
 import { getAvatarURL } from '../../api/api';
-import AvatarMenu from '@mui/material/Menu';
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:23]: Clicking the avatar no longer displays a menu.

File: web-ui/src/components/menu/Menu.jsx [hunk 2]
```diff
@@ -183,10 +181,6 @@ function Menu() {
     return links;
   };
 
-  const handleClick = event => {
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:186]: This is now handled by a `Link` component.

File: web-ui/src/components/menu/Menu.jsx [hunk 3]
```diff
@@ -304,10 +298,6 @@ function Menu() {
     setFeedbackOpen(false);
   };
 
-  const closeAvatarMenu = () => {
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:307]: There's no longer an avatar menu to close.

File: web-ui/src/components/menu/Menu.jsx [hunk 4]
```diff
@@ -430,14 +420,8 @@ function Menu() {
             <MenuIcon />
           </IconButton>
         </Toolbar>
-        <div
-          ref={anchorRef}
-          aria-controls={open ? 'menu-list-grow' : undefined}
-          aria-haspopup="true"
-          onClick={handleToggle}
-        >
+        <Link to={`/profile/${id}`}>
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:423]: The avatar is now wrapped in a `Link` that goes to the Profile page.

File: web-ui/src/components/menu/Menu.jsx [hunk 5]
```diff
@@ -430,14 +420,8 @@ function Menu() {
...
           <Avatar
-            onClick={handleClick}
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:440]: The click is now handled by the `Link` that wraps this `Avatar` component.

File: web-ui/src/components/menu/Menu.jsx [hunk 6]
```diff
@@ -447,42 +431,7 @@ function Menu() {
               textDecoration: 'none'
             }}
           />
-          <AvatarMenu
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:450]: No more avatar menu.

File: web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap [hunk 7]
```diff
@@ -33,8 +33,8 @@ exports[`<Menu /> > renders correctly 1`] = `
           />
         </button>
       </div>
-      <div
-        aria-haspopup="true"
+      <a
+        href="/profile/undefined"
```
Comment (mvolkmann) [inline web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap:37]: I wonder if the reason this contains "undefined" is that in the test we don't have a user id.

File: web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap [hunk 8]
```diff
@@ -436,8 +436,8 @@ exports[`<Menu /> > renders correctly for admin 1`] = `
...
```
Comment (mvolkmann) [inline web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap:440]: See the previous comment.

File: web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap [hunk 9]
```diff
@@ -911,8 +911,8 @@ exports[`<Menu /> > renders correctly for pdl 1`] = `
...
```
Comment (mvolkmann) [inline web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap:915]: See the previous comment.

**Review by S78901 — APPROVED** [review]

File: web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap (hunk 7 above)
Comment (vhscom) [reply web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap:37]: In `Menu.test.jsx` currently I see us taking snapshots, but I don't see us making any assertions yet. In this case, the `${id}` is set at runtime based on the user profile. If we want to test various render conditions, and just _that_ it renders, I think we have room to beef up our coverage.

**Review by vhscom — DISMISSED** [review]
:shipit:

File: web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap (hunk 7 above)
Comment (mvolkmann) [reply web-ui/src/components/menu/__snapshots__/Menu.test.jsx.snap]: I added a test to verify that the Link around the Avatar uses the id value.

File: web-ui/src/components/menu/Menu.test.jsx [hunk 10]
```diff
@@ -84,4 +88,19 @@ describe('<Menu />', () => {
...
+      <AppContextProvider value={initialState}>
+        <BrowserRouter>
+          <Menu />
+        </BrowserRouter>
+      </AppContextProvider>
+    );
+    const link = container.querySelector('header > a');
+    const href = link.getAttribute('href');
+    const lastIndex = href.lastIndexOf('/');
+    const id = href.substring(lastIndex + 1);
```
Comment (vhscom) [inline web-ui/src/components/menu/Menu.test.jsx:103]: Another way to grab the `id` could be to split the string into an array and pop off the last element:
```js
const id = href.split('/').pop()
```

**Review by vhscom — APPROVED** [review]
Love seeing some examples of how we're component testing. I saw we also have [Testing Library](https://testing-library.com/docs/react-testing-library/intro/), which can be useful as its API surface tends to nudge implementations towards semantic markup and use of ARIA where it matters most (e.g. component suspense states).

**Review by S78901 — APPROVED** [review]

**Pull request merged**